user-interface/src/main/java/life/qbic/datamanager/views/projects/create/ProjectDesignLayout.java

```diff
@@ -97,16 +128,32 @@ private void initCodeGenerationButton() {
     generateCodeButton.setTooltipText("Generate Project ID");
     generateCodeButton.addThemeVariants(ButtonVariant.LUMO_ICON);
     generateCodeButton.addClickListener(
-        buttonClickEvent -> codeField.setValue(ProjectCode.random().value()));
+        //Only non-used project code should be provided
+        buttonClickEvent -> {
+          String newProjectCode = ProjectCode.random().value();
+          boolean isProjectCodeDuplicated = !isProjectCodeUnique(newProjectCode);
+          while (isProjectCodeDuplicated) {
+            newProjectCode = ProjectCode.random().value();
+            isProjectCodeDuplicated = isProjectCodeUnique(newProjectCode);
+          }
+          codeField.setValue(newProjectCode);
+        });
   }
 
   private void initFieldValidators() {
     codeField.setRequired(true);
     titleField.setRequired(true);
     projectDescription.setRequired(true);
     projectDesignBinder.forField(codeField).withValidator(ProjectCode::isValid,
-            "A project code starts with Q followed by 4 letters/numbers").
-        bind(ProjectDesign::getProjectCode, ProjectDesign::setProjectCode);
+            "A project code starts with Q followed by 4 letters/numbers")
+        .withValidator((Validator<String>) (value, context) -> {
+          if (isProjectCodeUnique(value)) {
+            return ValidationResult.ok();
+          } else {
+            return ValidationResult.error(String.format("Project code %s is not unique", value));
```

Comment:
As this will be shown to the user, consider adding information on how to resolve the issue e.g. `The project code QABCD is already taken. Please choose a different code.`

Reply:
That's a very good idea 🧠